scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -364,6 +364,9 @@ function Initialize-IoTEdge {
 
     Set-ProvisioningMode
     Set-Certificates
+    if($DpsX509) {
```

Comment:
`$DpsX509` is optional. The cmdlet may be in the DPSX509 parameter set even without that particular switch. You need to look at `$PSCmdlet.ParameterSetName` just like line 282.

Also, we should do this for `ManualX509` too. Just in case.